hw02_unpack_string/unpack.go

```diff
@@ -2,11 +2,55 @@ package hw02_unpack_string //nolint:golint,stylecheck
 
 import (
 	"errors"
+	"log"
+	"strconv"
+	"strings"
+	"unicode"
 )
 
 var ErrInvalidString = errors.New("invalid string")
 
-func Unpack(_ string) (string, error) {
-	// Place your code here
-	return "", nil
+func Unpack(s string) (string, error) {
+	var b strings.Builder
+	var prevChar rune
+	// convert string to a slice of runes to properly handle multibyte chars
+	input := []rune(s)
+	for key, i := range input {
+		if key > 0 {
+			prevChar = input[key-1]
+		}
+
+		// linter is bs-ing with "deeply nested (complexity: 5)" here
```

Comment:
Довольно грубый комментарий. Лучше не писать такое даже в шутку, т.к. никогда не знаешь, кто увидит твой код.

Если линтер неправ, то это нормально и называется false-positive. Рекомендую почитать — https://golangci-lint.run/usage/false-positives. Хорошим тоном будет написать почему этот линтер отключен. Например, если бы все выходы из if'a были бы выходами из цикла с ошибкой (как первые два), то можно было бы написать: `//nolint:nestif // because nested ifs just returns error`.

В данном случае, можно довольно легко все упростить, если немного переработать алгоритм (об этом в другом дискашене).